QuizApp/PresentationLayer/ViewControllers/LoginViewController.swift

```diff
@@ -94,33 +98,43 @@ class LoginViewController: UIViewController {
         loginButton.addAction(.init {
             _ in
             self.errorLabel.isHidden = true
-            let status: LoginStatus = DataService().login(email: self.emailField.text!, password: self.passwordField.text!)
-            print(status)
-            switch(status){
-                case .success:
-                    self.router.showQuizzesScreen()
-                    break
-                case .error( _, _):
-                    self.errorLabel.isHidden = false
-                    break
+            
+            self.networkService.login(username: self.usernameField.text ?? "", password: self.passwordField.text ?? "") {
+                (result: Result<Bool, RequestError>) in
+                DispatchQueue.main.async {
+                    switch result {
+                    case .success(_):
+                        self.router.showQuizzesScreen()
+                        break
+                    
+                    case .failure(let error):
+                        print(error)
+                        self.errorLabel.isHidden = false
+                        break
+                    }
+                }
```

Comment:
It would be nice to extract this to a separate method, e.g. `handleLogin()` so the `buildViews()` doesn’t look so cluttered.